### DISCUSSION TITLE
Cardano shelley address is wrong.

### DISCUSSION THREAD

**Discussion by JasonLee-crypto** [trustwallet/wallet-core#1563] [Q&A]
**Describe the bug**
Use same recovery phrase 12 words, the address is different form `Exodus` wallet
wallet-core: `addr1sshjyujv0phjlmmrk8vqggzle4stnegkpuchhkc9qtull2ey5xn00g32394htkvxk3k4q5f30gzdwms8r85634aw8jdfa6yz4y64rsjrdsdltv`

Exodus:
`addr1qy2cfrarh6ya34t75n7syfgrh2ztv767za437yg5pf6jl2c4sj86805fmr2haf8aqgjs8w5ykea4u9mtrug3gzn4974svg3m0k`

Also note the former one is 115 chars while the later one is 103 chars

**To Reproduce**
use Exodus wallet to restore a 12 words wallet, check the address.
generate the address with the same 12 words with wallet-core

**Expected behavior**
should have the same address at least same length

**Screenshots**

**Additional context**

**Comment by optout21** [comment]
Could you provide some additional information to investigate:
- What is the address generated by Exodus for following test mnemonic: "cost dash dress stove morning robust group affair stomach vacant route volume yellow salute laugh"
- What is the derivation path used by Exodus?  Wallet core uses m/1852'/1815'/0'/0/0

**Comment by optout21** [comment]
I had a look at the provided Exodus address (`addr1qy2cfrarh6ya34t75n7syfgrh2ztv767za437yg5pf6jl2c4sj86805fmr2haf8aqgjs8w5ykea4u9mtrug3gzn4974svg3m0k`):
- it encodes 57 bytes, whereas wallet core uses/expects either 33 bytes (for Single, Account and Multisig addresses) or 65 bytes (for Group addresses)
- it contains a Kind value of 1, whereas valid values are between 3 -- 6.

I also found that wallet core uses Discrimination=1 value, which denotes test net.

More investigation needed.

**Comment by JasonLee-crypto** [comment]
@catenocrypt Exodus is an wallet app you can download from google play store or App store.
It support 12 words recovery phrase wallet recovery for now when I testing
It use the same derivation path ` m/1852'/1815'/0'/0/0`.

Also I know from cardano official site, its sheleey address contains `payment credential` & `staking credential`,
did you consider this part.
